packages/rule-parser/src/types.ts

```diff
@@ -0,0 +1,33 @@
+export type Value = string | number | boolean;
+export type Source = Value | Value[];
+export type Target = Value | Value[];
+export type Operator = string;
```

Comment:
Is it possible to type it as constants so that developers can get autocompletion?

Reply:
I'm not I understand this, can you provide an example of what this should change to?

Reply:
I believe the question is that rather than `string` can this be typed as a union of string literals: `'in' | '!in' | 'is' | …`.

Reply:
Ah yeah! It was originally like that, the issue I run into is that once you have an extensible registry that allows additional operators, the type would fail.

I suppose we can add a list of possible operators to the `parser< T >` call?

Reply:
I suppose the type can be
```suggestion
export type Operator =
	| 'is'
	| '='
	| 'not is'
	| '!='
	| 'contains'
	| 'not contains'
	| '!contains'
	| 'in'
	| 'not in'
	| '!in'
	| 'greater than'
	| '>'
	| 'less than'
	| '<'
	| 'gte'
	| '>='
	| 'lte'
	| '<='
	| string;
```